hummingbot/connector/exchange/bittrex/bittrex_api_user_stream_data_source.py

```diff
@@ -1,160 +1,112 @@
-        self._bittrex_auth: BittrexAuth = bittrex_auth
-        self._trading_pairs = trading_pairs
-        self._current_listen_key = None
-        self._listen_for_user_stream_task = None
-        self._last_recv_time: float = 0
-        self._websocket_connection: Optional[signalr_aio.Connection] = None
-        self._hub = None
+    def __init__(self, auth: BittrexAuth,
+                 connector,
+                 api_factory):
         super().__init__()
-
-    @property
-    def hub(self):
-        return self._hub
-
-    @hub.setter
-    def hub(self, value):
-        self._hub = value
+        self._auth = auth
+        self._connector = connector
+        self._api_factory = api_factory
+        self._websocket_connection: Optional[signalr_aio.Connection] = None
+        self.hub = None
 
     @property
     def last_recv_time(self) -> float:
-        return self._last_recv_time
+        """
+        Returns the time of the last received message
+
+        :return: the timestamp of the last received message in seconds
+        """
+        if self._ws_assistant:
+            return self._auth.time_provider.time()
+        return 0
+
+    async def _get_ws_assistant(self) -> WSAssistant:
+        if self._ws_assistant is None:
+            self._ws_assistant = await self._api_factory.get_ws_assistant()
+        return self._ws_assistant
+
+    async def _connected_websocket_assistant(self):
+        websocket_connection = signalr_aio.Connection(CONSTANTS.BITTREX_WS_URL, session=None)
+        self.hub = websocket_connection.register_hub("c3")
+        auth_params = self._auth.generate_WS_auth_params()
+        self.hub.server.invoke("Authenticate", auth_params)
+        return websocket_connection
+
+    # TODO Remove this code after verification
+    '''async def _authenticate_client(self, ws: WSAssistant):
+        try:
+            ws_request: WSJSONRequest = WSJSONRequest(
+                {
+                    "H": "c3",
+                    "M": "Authenticate",
+                    "A": [],
+                    "I": 1
+                }
+            )
+            auth_params = self._auth.generate_WS_auth_params()
+            ws_request.payload['A'] = auth_params
+            await ws.send(ws_request)
+            resp: WSResponse = await ws.receive()
+            auth_response = resp.data["R"]
+            if not auth_response["Success"]:
+                raise ValueError(f"User Stream Authentication Fail! {auth_response['ErrorCode']}")
+            self.logger().info("Successfully authenticated to user stream...")
+        except asyncio.CancelledError:
+            raise
+        except Exception:
+            self.logger().error("Error occurred authenticating websocket connection..")
+            raise'''
+
+    async def _subscribe_channels(self, websocket_assistant):
+        try:
+            self.hub.server.invoke("Subscribe", ["heartbeat", "order", "balance", "execution"])
```

Comment:
I see that we are subscribed to the "heartbeat" channel, but it doesnt seem like we're doing anything with it

Reply:
Removed subscription to heartbeat